.github/workflows/docker-build.yml

```
@@ -21,40 +22,41 @@ jobs:
     permissions:
       contents: write
       packages: write
-
     steps:
       - uses: "actions/checkout@v3"
-
-      - name: set up go
-        uses: actions/setup-go@v3
-        with:
-          go-version: ${{ env.GO_VERSION }}
-
       - name: Docker meta
         id: meta
         uses: docker/metadata-action@v4
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
           tags: |
             type=sha
-            type=semver,pattern={{major}}.{{minor}}.{{patch}}
-            type=semver,pattern={{major}}.{{minor}}
-            type=semver,pattern={{major}}
-
+            type=semver,pattern={{version}}
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
-
       - name: Login to GHCR
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Build and push
+      - name: Build and Push amd64 Image
+        if: |
+          github.ref != 'refs/heads/main' &&
+          !startsWith(github.ref, 'refs/tags/v')
+        uses: docker/build-push-action@v3
+        with:
+          platforms: linux/amd64
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          file: docker/Dockerfile
+      - name: Build and Push amd64+arm64 Images
+        if: |
+          github.ref == 'refs/heads/main' ||
+          startsWith(github.ref, 'refs/tags/v')
```

With the changes to the triggers, we can just check for the push event type because we are already limiting to push events on `main` and `v*` tags
```suggestion
        if: ${{ github.event_name == 'push' }}
```